Classes/Domain/Repository/EmployeeRepository.php

```diff
@@ -21,6 +26,43 @@
  */
 class EmployeeRepository extends Repository
 {
+    protected GlossaryService $glossaryService;
+
+    public function injectGlossaryService(GlossaryService $glossaryService): void
+    {
+        $this->glossaryService = $glossaryService;
+    }
+
+    public function findFilteredBy($office, array $search = []): QueryResultInterface
+    {
+        $query = $this->createQuery();
+        $constraints = [];
+
+        if ($office) {
+            $constraints[] = $query->equals('office', $office);
+        }
+
+        if (((string)$search['letter'] ?? '') !== '') {
+            $constraints[] = $this->glossaryService->getLetterConstraintForExtbaseQuery(
+                $query,
+                'lastName',
+                $search['letter'],
+            );
+        }
+
+        if (count($constraints) === 1) {
+            $query->matching(reset($constraints));
+        } elseif (count($constraints) >= 2) {
+            $query->matching($query->logicalAnd(...$constraints));
+        }
+
+        if ($constraints === []) {
+            return $query->execute();
+        }
+
+        return $query->matching($query->logicalAnd(...$constraints))->execute();
```

Comment:
If $constraints has 3 array entries, it is greater than 2. So logicalAnd will be filled with the array. OK.
Row 59 is not true in that case.
In row 63 you add all constraints to logicalAnd to query again.